cshared.go

```diff
@@ -325,3 +376,7 @@ func makeMetrics(cmp *cmetrics.Context) Metrics {
 		},
 	}
 }
+
+func GetLogger() Logger {
+	return logger
```

Comment:
I wouldn't make this a package level thing.
Maybe `Init()` method on the plugin could receive it in its args.

Reply:
`Init()` is already receiving too many args.

```go
ctx context.Context
conf plugin.ConfigLoader
metrics plugin.Metrics
```

And adding `plugin.Logger` to those... Maybe we can pass a struct with all of those in there.

```go
type Fluentbit struct {
  Conf plugin.ConfigLoader
  Metrics plugin.Metrics
  Logger plugin.Logger
}
```

`Init()` can now looks like this:

```go
func (*myPlugin) Init(ctx context.Context, fbit plugin.Fluentbit) error {
  fbit.Config.String("some_param")
  fbit.Metrics.NewCounter("my_counter", "desc")
  fbit.Logger.Info("calling init")

  return nil
}

Reply:
Looks good. I implemented your suggestion. ¡Genial!